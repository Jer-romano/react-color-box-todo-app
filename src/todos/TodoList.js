import {React, useState} from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import {v4 as uuid } from 'uuid';

const TodoList = () => {
    const [todos, setTodos] = useState([]);

    const renderTodos = () => {
        return (
            <div>
                {todos.map(t => (<Todo id={t.id}
                                       key={t.id}
                                       text={t.text}
                                       removeTodo={removeTodo}
                                 />))}
            </div>
        );
    };

    const addTodo = (todo) => {
        let newTodo = { ...todo, id: uuid() };
        setTodos(todos => [...todos, newTodo]);
    };

    const removeTodo = (e) => {
        let id = e.target.id;
        const result = todos.filter(t => t.id !== id);
        setTodos(result);
    }

    return (
        <div className='TodoList'>
            <NewTodoForm addTodo={addTodo}/>
            {renderTodos()}
        </div>
    );

};

export default TodoList;