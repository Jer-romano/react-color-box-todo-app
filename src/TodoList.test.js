import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import TodoList from './todos/TodoList';

it("renders without crashing", function() {
    render(<TodoList/>);
});

it("matches snapshot", function() {
    const { asFragment } = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot();
});

it("can add a todo", function() {
    const {getByLabelText, queryByText} = render(<TodoList/>);

    expect(queryByText("Do laundry")).not.toBeInTheDocument();

    const todoInput = getByLabelText("New Todo:");
    const submitBtn = queryByText("Add Todo");

    fireEvent.change(todoInput, {target: {value: "Do laundry"}});
    fireEvent.click(submitBtn);

    expect(queryByText("Do laundry")).toBeInTheDocument();
    
});

it("can add and delete a todo", function() {
    const {getByLabelText, queryByText, getByText} = render(<TodoList/>);

    expect(queryByText("Do homework")).not.toBeInTheDocument();

    const todoInput = getByLabelText("New Todo:");
    const submitBtn = queryByText("Add Todo");

    fireEvent.change(todoInput, {target: {value: "Do homework"}});
    fireEvent.click(submitBtn);

    expect(queryByText("Do homework")).toBeInTheDocument();

    const deleteBtn = getByText("X");

    fireEvent.click(deleteBtn);
    expect(queryByText("Do homework")).not.toBeInTheDocument();
    
});