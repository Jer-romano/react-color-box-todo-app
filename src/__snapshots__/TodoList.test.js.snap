// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <form>
      <label
        for="text"
      >
        New Todo:
      </label>
      <input
        id="text"
        name="text"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
